src/pkg/services/m365/api/drive_pager_test.go

```diff
@@ -210,3 +214,226 @@ func (suite *DrivePagerIntgSuite) TestEnumerateDriveItems() {
 	require.NotEmpty(t, items, "should find items in user's drive")
 	assert.NotEmpty(t, du.URL, "should have a delta link")
 }
+
+// TestDriveDeltaPagerQueryParams is a regression test to check if
+// the delta pager is setting query params and prefer headers as
+// expected.
+func (suite *DrivePagerIntgSuite) TestDriveDeltaPagerQueryParams() {
+	tests := []struct {
+		name        string
+		setupf      func()
+		forURLCache bool
+		expect      assert.ErrorAssertionFunc
+	}{
+		// Separate out tests for $select and $prefer because otherwise
+		// it's hard to tell which gock check is failing.
+		{
+			name: "check $top",
+			setupf: func() {
+				delta := msDrive.NewItemItemsItemDeltaResponse()
+				delta.SetValue([]models.DriveItemable{
+					models.NewDriveItem(),
+				})
+
+				str := "deltaLink"
+				delta.SetOdataDeltaLink(&str)
+
+				interceptV1Path("drives", "drive", "items", "root", "delta()").
+					MatchParam("$top", strconv.Itoa(int(maxDeltaPageSize))).
+					Reply(200).
+					JSON(graphTD.ParseableToMap(suite.T(), delta))
+			},
+			expect: assert.NoError,
+		},
+		{
+			name: "check $select",
+			setupf: func() {
+				delta := msDrive.NewItemItemsItemDeltaResponse()
+				delta.SetValue([]models.DriveItemable{
+					models.NewDriveItem(),
+				})
+
+				str := "deltaLink"
+				delta.SetOdataDeltaLink(&str)
+
+				// This is a copy of DefaultDriveItemProps. This list should be
+				// adjusted while changing DefaultDriveItemProps. This however
+				// doesn't protect against adding new params to DefaultDriveItemProps
+				// since gock doesn't allow asserting for presence of unexpected
+				// params.
+				params := idAnd(
+					"content.downloadUrl",
+					"createdBy",
+					"createdDateTime",
+					"file",
+					"folder",
+					"lastModifiedDateTime",
+					"name",
+					"package",
+					"parentReference",
+					"root",
+					"sharepointIds",
+					"size",
+					"deleted",
+					"malware",
+					"shared")
+
+				selectParams := params[0]
+				for _, p := range params[1:] {
+					selectParams += "," + p
+				}
```

Comment:
nit: could I interest you in the `strings.Join` function? (there's some similar chunks of code in this PR that I haven't called out directly)
```suggestion
				selectParams := strings.Join(params, ",")
```

Reply:
Thanks! Fixed.